builtin/logical/pki/cert_util.go

```diff
@@ -644,21 +650,29 @@ func generateCreationBundle(b *backend,
 	ipAddresses := []net.IP{}
 	var ipAltInt interface{}
 	{
-		ipAltInt, ok = data.GetOk("ip_sans")
-		if ok {
-			ipAlt := ipAltInt.(string)
-			if len(ipAlt) != 0 {
-				if !role.AllowIPSANs {
-					return nil, errutil.UserError{Err: fmt.Sprintf(
-						"IP Subject Alternative Names are not allowed in this role, but was provided %s", ipAlt)}
-				}
-				for _, v := range strings.Split(ipAlt, ",") {
-					parsedIP := net.ParseIP(v)
-					if parsedIP == nil {
+		if csr != nil && role.UseCSRSANs {
+			if !role.AllowIPSANs {
+				return nil, errutil.UserError{Err: fmt.Sprintf(
+					"IP Subject Alternative Names are not allowed in this role, but was provided some via CSR")}
+			}
+			ipAddresses = csr.IPAddresses
+		} else {
```

Comment:
Yes, if the CSR is non-nil but UseCSRSANs isn't set, we want to use the supplied request values.